src/pages/SwapV3/index.tsx

```diff
@@ -0,0 +1,567 @@
+    if (showingPairSuggestionImport) {
+      setShowingPairSuggestionImport(false)
+    } else {
+      setDismissTokenWarning(true)
+    }
+  }, [showingPairSuggestionImport])
+
+  const handleConfirmTokenWarning = useCallback(
+    (tokens: Currency[]) => {
+      handleDismissTokenWarning()
+      if (showingPairSuggestionImport) {
+        refSuggestPair.current?.onConfirmImportToken() // callback from children
+      }
+      if (isLimitPage) {
+        onSelectPairLimit(tokens[0], tokens[1])
+      }
+    },
+    [isLimitPage, onSelectPairLimit, showingPairSuggestionImport, handleDismissTokenWarning],
+  )
+
+  const handleRotateClick = useCallback(() => {
+    onSwitchTokensV2()
+    setIsSelectCurrencyManually(true)
+  }, [onSwitchTokensV2])
+
+  // TODO: fix mixpanel
+  const { mixpanelHandler } = useMixpanel(undefined, currencies)
+
+  const onSelectSuggestedPair = useCallback(
+    (fromToken: Currency | undefined, toToken: Currency | undefined, amount?: string) => {
+      if (isLimitPage) {
+        onSelectPairLimit(fromToken, toToken)
+        return
+      }
+      if (fromToken) onCurrencySelection(Field.INPUT, fromToken)
+      if (toToken) onCurrencySelection(Field.OUTPUT, toToken)
+      if (amount) handleTypeInput(amount)
+    },
+    [handleTypeInput, onCurrencySelection, onSelectPairLimit, isLimitPage],
+  )
+
+  const tokenImports: Token[] = useUserAddedTokens()
+  const prevTokenImports = usePrevious(tokenImports)
+
+  useEffect(() => {
+    // when remove token imported
+    if (!prevTokenImports) return
+    const isRemoved = prevTokenImports?.length > tokenImports.length
+    if (!isRemoved || prevTokenImports[0].chainId !== chainId) return
+
+    const addressIn = currencyIn?.wrapped?.address
+    const addressOut = currencyOut?.wrapped?.address
+    // removed token => deselect input
+    const tokenRemoved = prevTokenImports.filter(
+      token => !tokenImports.find(token2 => token2.address === token.address),
+    )
+
+    tokenRemoved.forEach(({ address }: Token) => {
+      if (address === addressIn || !currencyIn) {
+        onResetSelectCurrency(Field.INPUT)
+      }
+      if (address === addressOut || !currencyOut) {
+        onResetSelectCurrency(Field.OUTPUT)
+      }
+    })
+  }, [tokenImports, chainId, prevTokenImports, currencyIn, currencyOut, onResetSelectCurrency])
+
+  const isLoadedTokenDefault = useIsLoadedTokenDefault()
+
+  useEffect(() => {
+    if (isExpertMode) {
+      mixpanelHandler(MIXPANEL_TYPE.ADVANCED_MODE_ON)
+    }
+  }, [isExpertMode, mixpanelHandler])
+
+  const [rawSlippage, setRawSlippage] = useUserSlippageTolerance()
+
+  const isStableCoinSwap =
+    INPUT?.currencyId &&
+    OUTPUT?.currencyId &&
+    chainId &&
+    STABLE_COINS_ADDRESS[chainId].includes(INPUT?.currencyId) &&
+    STABLE_COINS_ADDRESS[chainId].includes(OUTPUT?.currencyId)
+
+  const rawSlippageRef = useRef(rawSlippage)
+  rawSlippageRef.current = rawSlippage
+
+  useEffect(() => {
```

Comment:
rawSlippage
 isStableCoinSwap 
 rawSlippageRef
 
sao cứ thấy 3 thằng này phụ thuộc lẫn nhau mà chả có thằng nào khác dùng

Reply:
They're only used in that useEffect to update the slippage. I removed this piece of code as already had a hook